### ISSUE TITLE
mas wants to update to versions that do not exist on the app store

### ISSUE THREAD

**Issue by dmuiX** [mas-cli/mas#515]
## Your Environment

- `mas version`: 1.8.6
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 10.1...

      System Version: macOS 14.5 (23F79)
      Kernel Version: Darwin 23.5.0

### mas Install Method

- [] `brew install mas` (homebrew-core)
- [x] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
  - Fork/branch: ? (e.g. mas-cli/main)
  - Xcode version: 10.?

## Describe the Bug

> A clear and concise description of what the bug is.

if i enter `mas outdated` its showing me 

897283731 Strongbox (1.59.11 -> 1.59.12)
1500855883 CapCut (3.8.0 -> 11.8.0)
310633997 WhatsApp (24.11.84 -> 24.12.78)

if I do `mas upgrade` it upgrades my apps everytime although they cannot be updated...
seems like mas is using a different app store then my macbook...maybe some weird beta versions?? 

On the appstore strongbox is 1.59.11

<img width="1185" alt="Screenshot 2024-06-18 at 13 25 00" src="https://github.com/mas-cli/mas/assets/19862760/b3a909b3-3ec5-47bd-8c46-11d36647c9fd">

capcut is 3.8.0 

<img width="1185" alt="Screenshot 2024-06-18 at 13 29 18" src="https://github.com/mas-cli/mas/assets/19862760/99f4b798-587d-4113-8db3-6883efc15a6e">

whatsapp is 24.11.84

<img width="1175" alt="Screenshot 2024-06-18 at 13 29 50" src="https://github.com/mas-cli/mas/assets/19862760/18bcb435-da1e-4475-9454-f917091d44fa">

## To Reproduce

mas outdated
mas update

## Expected Behavior

> A clear and concise description of what you expected to happen.

mas should leave apps on the level that is in the appstore 

## Actual Behavior

> A clear and concise description of what actually happened.

mas wants to upgrade my apps everytime I run `mas upgrade` but it shouldnt do that. 

## Screenshots, Terminal Output

> If applicable, add screenshots to help explain your problem.

```bash
$ mas command
...
```

## Additional Context

> Add any other context about the problem here.
> Did it work in a previous version?

**Comment by dmuiX** [comment]
Seems this is a dup, which might be solved in current beta.
How can I download that one? Currently just zip packages no pkg available

**Comment by rgoldberg** [comment]
@dmuiX  Can we close this as a duplicate of #491?

The following should be the current 1.8.7 beta. It has a `.pkg`:

https://github.com/mas-cli/mas/releases/tag/v1.8.7-beta.33

**Comment by andmpel** [comment]
I tried the latest beta and I'm experiencing this issue

**Comment by rgoldberg** [comment]
@andmpel How did you install the latest beta?

Please provide exact actions, like ran installer dowloaded from `https://downloads.example.com/…`. Got that URL from web page `https://example.com/…`. Or cloned repository using `gh clone…`, … etc.

**Comment by andmpel** [comment]
I installed the latest prerelease from GitHub releases using the .pkg asset

**Comment by rgoldberg** [comment]
@andmpel Did you install from the following URL?

https://github.com/mas-cli/mas/releases/download/v1.8.7-beta.33/mas.pkg

**Comment by andmpel** [comment]
I was mistaken, I had to remove the brew mas formula for the new version to be used.

Thanks.

**Comment by andmpel** [comment]
Is this package even being maintained?

**Comment by rgoldberg** [comment]
I am trying to revive it. I've only done a tiny bit of work on it before. The project lead has been sporadically working on 1.8.7 beta. I'm trying to get in touch to see what else must be done before releasing 1.8.7, to learn if all the revised release procedures, etc. are documented, and to see how he wants to proceed on a few different issues (like support for older versions of macOS that are no longer supported by Apple, priorities for new development, choosing between alternative solutions for issues, etc.). So, hopefully things will improve soon.

**Comment by rgoldberg** [comment]
Closing as a duplicate of #491. PR #496 fixed the issue. It will be released in the upcoming version 1.8.7, which is currently in beta.

If you think this issue is different than #491, please let us know.
